### PR TITLE
select sources by table

### PR DESCRIPTION
**Pull request by pcuste1**
## What is changing?
Enabling selection of sources in aladin from an astropy table by unique key(s)

## Why is this change needed?
This change enables both [CST-171](https://jira.stsci.edu/browse/CST-171) and [CST-179](https://jira.stsci.edu/browse/CST-179) by enabling selection of sources defined in an astropy table, and by supporting matching by a user defined set of keys. 

This will enable automation of notebooks in a variety of ways. For our initial use case, this will enable importing and exporting data to and from ipyaladin and jdaviz. 

## How was this tested?
The attached notebook was used to manually test these changes. In the future it would be helpful to have an automated test suite that could be used to test a variety of edge cases.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,7 +24,8 @@ docs = ["autoapi",
     "sphinx-collections",
     "sphinx-copybutton",
     "sphinx-gallery",
-    "pydata-sphinx-theme"
+    "pydata-sphinx-theme",
+    "pandas"
```
Comment (pcuste1) [inline pyproject.toml:28]: note: we had to add pandas here since we are converting the astropy table to pandas in order to get the json safe dictionary of user defined keys to search on. This is the best way I've found to do this, but I'm open to any suggestions

File: js/models/message_handler.js [hunk 2]
```diff
@@ -155,6 +155,27 @@ export default class MessageHandler {
...
+        }
+        return cat.getSources().filter((source) =>
+          sources.some((s) =>
+            keys.every((key) => {
+              return s[key] == source.data[key];
+            }),
+          ),
+        );
+      })
+      .filter((n) => n.length);
```
Comment (pcuste1) [inline js/models/message_handler.js:174]: note: this filter statement is here to remove any empty lists that may be returned from the first filter statement. If there are multiple catalogs, and the key only matches on some of them, it will return something like the following
```
[
    [],     // empty list
    [ ... ] // list full of sources
]
```
The empty list needs to be removed for the `selectObjects` call to function properly

File: js/models/message_handler.js [hunk 3]
```diff
@@ -155,6 +155,27 @@ export default class MessageHandler {
     this.aladin.select(selectionType);
   }
 
+  handleTriggerSelectionByTable(msg) {
+    let sources = msg["table"];
+    let keys = msg["keys"];
+    let sourcesByCatalog = this.aladin.view.catalogs
+      .map((cat) => {
+        if (!cat.isShowing) {
+          return;
```
Comment (tomdonaldson) [inline js/models/message_handler.js:164]: Might be clearer to explicitly return `[]`.

File: js/models/message_handler.js [hunk 4]
```diff
@@ -155,6 +155,27 @@ export default class MessageHandler {
...
+        }
+        return cat.getSources().filter((source) =>
+          sources.some((s) =>
```
Comment (tomdonaldson) [inline js/models/message_handler.js:167]: I found the name `sources` confusing here since they are very different from the `getSources()` and the `source` being filtered.  Perhaps rename to indicate that they are the sources to be selected.  This could tie in with a succinct comment or two here that clarify how the nested functions are filtering the lists.

I'd also be tempted to store the `filter()` result in a local variable and return that on a separate line.  This could help the readability a little and would make any interactive debugging easier.

File: js/models/message_handler.js (hunk 2 above)
Comment (tomdonaldson) [reply js/models/message_handler.js:174]: I feel like there is already enough happening with all the nested functions in the `map()` line.  I would just do this last `filter()` on a new line preceded by a comment.
